src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -409,10 +409,35 @@ export const DatasetBuilderContents = ({ onStateChange }: { onStateChange: OnSta
       }),
       h(ConceptSetSelector, {
         selectedConceptSets,
-        onChange: (conceptSets) => setSelectedConceptSets(conceptSets),
+        onChange: async (conceptSets) => {
+          setSelectedConceptSets(conceptSets);
+          const uniqueFeatureValueGroups = _.flow(
+            _.flatMap((headerAndValues: HeaderAndValues<ConceptSet>) => headerAndValues.values),
+            _.map((conceptSet: ConceptSet) => conceptSet.featureValueGroupName),
+            _.uniq
+          )(conceptSets);
+          setValues(
+            _.flow(
+              _.filter((featureValueGroup: FeatureValueGroup) =>
```

Comment:
I'm surprised that you need to explicitly type the arguments here (e.g. `headerAndValues` and `conceptSet`) I would have thought that type inference would be sufficient.  Or is that a pattern in the repo to always specify the type?

Reply:
It has to do with `_.flow` not really coercing types well, unfortunately. So with flow you often end up needing to explicitly type.